CTestGameWorld.ClosestCharacter fails in s390x

On Fedora (40 up to 43/rawhide), building DDNet 19.2 in the s390x architecture fails in CTestGameWorld.ClosestCharacter with the following error. No other architecture presented this issue.

```
[ RUN      ] CTestGameWorld.ClosestCharacter
2025-05-12 20:25:49 I storage: added path 'CTestGameWorld.ClosestCharacter-9934.tmp'
2025-05-12 20:25:49 I storage: added path '$DATADIR' ('data')
2025-05-12 20:25:49 I storage: added path '$CURRENTDIR' ('/builddir/build/BUILD/ddnet-19.2-build/ddnet-19.2/redhat-linux-build')
2025-05-12 20:25:49 I host_lookup: host='localhost' port=0 1
2025-05-12 20:25:49 I host_lookup: host='localhost' port=0 2
malloc(): invalid size (unsorted)
FAILED: CMakeFiles/run_cxx_tests /builddir/build/BUILD/ddnet-19.2-build/ddnet-19.2/redhat-linux-build/CMakeFiles/run_cxx_tests 
cd /builddir/build/BUILD/ddnet-19.2-build/ddnet-19.2/redhat-linux-build && /builddir/build/BUILD/ddnet-19.2-build/ddnet-19.2/redhat-linux-build/testrunner
ninja: build stopped: subcommand failed.
```

See [build.log](https://github.com/user-attachments/files/20179872/build.log) and [whole build task in Koji](https://koji.fedoraproject.org/koji/taskinfo?taskID=132649417).

This build was done in the build server, which I don't have access to the test logs. I tried to set a local environment to build and get the logs, but I still did not have success. I hope this report helps anyway.

--- a/src/engine/shared/datafile.cpp
+++ b/src/engine/shared/datafile.cpp
@@ -625,7 +625,10 @@ bool CDataFileReader::Open(class IStorage *pStorage, const char *pFilename, int 
 		return false;
 	}
 
-	SwapEndianInPlace(pTmpDataFile->m_pData, pTmpDataFile->m_Header.m_Swaplen);
+	// The swap len also includes the size of the header (without the size offset), but the header was already swapped above.
+	const int64_t DataSwapLen = pTmpDataFile->m_Header.m_Swaplen - (int)(sizeof(Header) - Header.SizeOffset());
+	dbg_assert(DataSwapLen == Size, "Swap len and file size mismatch");
+	SwapEndianInPlace(pTmpDataFile->m_pData, DataSwapLen);
 
 	pTmpDataFile->m_Info.m_pItemTypes = (CDatafileItemType *)pTmpDataFile->m_pData;
 	pTmpDataFile->m_Info.m_pItemOffsets = (int *)&pTmpDataFile->m_Info.m_pItemTypes[pTmpDataFile->m_Header.m_NumItemTypes];

